docs/architecture/adr-047-handling-evidence-from-light-client.md

```diff
@@ -5,210 +5,186 @@
 * 24-02-2020: Second version
 * 13-04-2020: Add PotentialAmnesiaEvidence and a few remarks
 * 31-07-2020: Remove PhantomValidatorEvidence
-* 14-08-2020: Introduce light traces
+* 14-08-2020: Introduce light traces (listed now as an alternative approach)
+* 20-08-2020: Light client produces evidence when detected instead of passing to full node
+* 16-09-2020: Post-implementation revision
+
+### Glossary of Terms
+
+- a `LightBlock` is the unit of data that a light client receives, verifies and stores.
+It is composed of a validator set, commit and header all at the same height.
+- a **Trace** is seen as an array of light blocks across a range of heights that were
+created as a result of skipping verification.
+- a **Provider** is a full node that a light client is connected to and serves the light
+client signed headers and validator sets.
+- `VerifySkipping` (sometimes known as bisection or verify non-adjacent) is a method the
+light client uses to verify a target header from a trusted header. The process involves verifying
+intermediate headers in between the two by making sure that 1/3 of the validators that signed
+the trusted header also signed the untrusted one.
+- **Light Bifurcation Point**: If the light client was to run `VerifySkipping` with two providers
```

Comment:
Doesn't this assume that the traces have headers at the same height?

Reply:
Imagine we have this function called `verifySkipping` that we give it a target height that it verifies to. The output is an array of light blocks that were verified on the way and the last of which is the one of that target height. When we compare this against a witness we run a for loop over each of these intermediate light blocks and for each one we do `verifySkipping`. As you alluded to for some of these heights, the witness won't be able to directly jump to (because the validator sets are too different) but then it will produce more intermediate headers (between the primary's witness headers) as part of verification (but unless the block hasn't actually been produced yet it should always find a way to bisect to those intermediate heights). I hope this explains things with a bit more clarity. If you think I should be adding more of an explanation about this in this text, then let me know.